src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/TableSortClauseBuilder.java

```diff
@@ -1,21 +1,79 @@
 package io.stargate.sgv2.jsonapi.api.model.command.builders;
 
+import static io.stargate.sgv2.jsonapi.exception.ErrorFormatters.errFmtApiColumnDef;
+import static io.stargate.sgv2.jsonapi.exception.ErrorFormatters.errFmtCqlIdentifier;
+import static io.stargate.sgv2.jsonapi.exception.ErrorFormatters.errVars;
 import static io.stargate.sgv2.jsonapi.util.JsonUtil.arrayNodeToVector;
 
+import com.datastax.oss.driver.api.core.CqlIdentifier;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.sort.SortClause;
 import io.stargate.sgv2.jsonapi.api.model.command.clause.sort.SortExpression;
+import io.stargate.sgv2.jsonapi.exception.SortException;
 import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiColumnDef;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiColumnDefContainer;
+import io.stargate.sgv2.jsonapi.util.CqlIdentifierUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
 /** {@link SortClauseBuilder} to use with Tables. */
 public class TableSortClauseBuilder extends SortClauseBuilder<TableSchemaObject> {
+  private record SortExpressionDefinition(String path, JsonNode sortValue, ApiColumnDef column) {}
+
   public TableSortClauseBuilder(TableSchemaObject table) {
     super(table);
   }
 
   @Override
-  protected SortExpression buildSortExpression(
-      String path, JsonNode innerValue, int sortExpressionCount) {
+  protected SortClause buildClauseFromDefinition(ObjectNode sortNode) {
+    // First, resolve the paths to column definitions
+    final List<SortExpressionDefinition> sortExprDefs = resolveColumns(sortNode);
```

Comment:
nit: not sure final is needed

Reply:
It is not strictly required, no. Matter of style I suppose (some java devs make everything possible final).

Reply:
Removed.